tools/autodeployment/webhook.js

```diff
@@ -44,7 +45,8 @@ function requestFilter(repo, buildType, action, ref, mainBranch) {
     // Check if the reference is the branch used as main or if it is a tag
     (currentBranch === mainBranch || ref.includes('tags/')) &&
     // Check the build type and the action
-    (buildType === 'staging' || (buildType === 'production' && action === 'published'))
+    (buildType === 'staging' || (buildType === 'production' && action === 'published')) &&
+    headCommit === 'master'
```

Comment:
I don't understand this.  The `head_commit` is going to be an object that looks like this (see https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#webhook-payload-example-2):

```json
  "head_commit": {
    "id": "6113728f27ae82c7b1a177c8d03f9e96e0adf246",
    "tree_id": "4b825dc642cb6eb9a060e54bf8d69288fbee4904",
    "distinct": true,
    "message": "Adding a .gitignore file",
    "timestamp": "2019-05-15T15:20:41Z",
    "url": "https://github.com/Codertocat/Hello-World/commit/6113728f27ae82c7b1a177c8d03f9e96e0adf246",
    "author": {
      "name": "Codertocat",
      "email": "21031067+Codertocat@users.noreply.github.com",
      "username": "Codertocat"
    }
```

Reply:
@tcvan0707 Can we look into what @manekenpix suggested on the issue, to use `target_commitish` from the [webhook](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#webhook-payload-example-36) from GitHub? The payload we get to grab the `target_commitish` is found on [this line](https://github.com/Seneca-CDOT/telescope/blob/cfc8f8f30f67869715f3dceed836cd76c4c76275/tools/autodeployment/webhook.js#L66)

Reply:
We can move the const reference to be global so you can grab reference outside of the function(but make sure its valid before we use it). We can try a build with this to test it, let us know when you are ready!